pkg/kubelet/kubelet_cadvisor_test.go

```diff
@@ -24,44 +24,183 @@ import (
+		requestedPodFullName      string
+		requestedPodUid           types.UID
+		requestedContainerName    string
+		expectDockerContainerCall bool
+		mockError                 error
+		expectedError             error
+		expectStats               bool
+	}{
+		{
+			name:          "get container info",
+			containerID:   "ab2cdf",
+			containerPath: "/docker/ab2cdf",
+			cadvisorContainerInfo: cadvisorapi.ContainerInfo{
+				ContainerReference: cadvisorapi.ContainerReference{
+					Name: "/docker/ab2cdf",
+				},
+			},
+			runtimeError: nil,
+			podList: []*kubecontainertest.FakePod{
+				{
+					Pod: &kubecontainer.Pod{
+						ID:        "12345678",
+						Name:      "qux",
+						Namespace: "ns",
+						Containers: []*kubecontainer.Container{
+							{
+								Name: "foo",
+								ID:   kubecontainer.ContainerID{Type: "test", ID: "ab2cdf"},
+							},
+						},
+					},
+				},
+			},
+			requestedPodFullName:      "qux_ns",
+			requestedPodUid:           "",
+			requestedContainerName:    "foo",
+			expectDockerContainerCall: true,
+			mockError:                 nil,
+			expectedError:             nil,
+			expectStats:               true,
 		},
-	}
-
-	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
-	fakeRuntime := testKubelet.fakeRuntime
-	kubelet := testKubelet.kubelet
-	cadvisorReq := &cadvisorapi.ContainerInfoRequest{}
-	mockCadvisor := testKubelet.fakeCadvisor
-	mockCadvisor.On("DockerContainer", containerID, cadvisorReq).Return(containerInfo, nil)
-	fakeRuntime.PodList = []*kubecontainertest.FakePod{
-		{Pod: &kubecontainer.Pod{
-			ID:        "12345678",
-			Name:      "qux",
-			Namespace: "ns",
-			Containers: []*kubecontainer.Container{
+		{
+			name:                  "get container info when cadvisor failed",
+			containerID:           "ab2cdf",
+			containerPath:         "/docker/ab2cdf",
+			cadvisorContainerInfo: cadvisorapi.ContainerInfo{},
+			runtimeError:          nil,
+			podList: []*kubecontainertest.FakePod{
 				{
-					Name: "foo",
-					ID:   kubecontainer.ContainerID{Type: "test", ID: containerID},
+					Pod: &kubecontainer.Pod{
+						ID:        "uuid",
+						Name:      "qux",
+						Namespace: "ns",
+						Containers: []*kubecontainer.Container{
+							{
+								Name: "foo",
+								ID:   kubecontainer.ContainerID{Type: "test", ID: "ab2cdf"},
+							},
+						},
+					},
 				},
 			},
-		}},
-	}
-	stats, err := kubelet.GetContainerInfo("qux_ns", "", "foo", cadvisorReq)
-	if err != nil {
-		t.Errorf("unexpected error: %v", err)
+			requestedPodFullName:      "qux_ns",
+			requestedPodUid:           "uuid",
+			requestedContainerName:    "foo",
+			expectDockerContainerCall: true,
+			mockError:                 cadvisorApiFailure,
+			expectedError:             cadvisorApiFailure,
+			expectStats:               false,
+		},
+		{
+			name:                      "get container info on non-existent container",
+			containerID:               "",
```

Comment:
optional: you don't need to specify default values (`""`, `false`, `nil`), so they can be omitted to shorten this. If you think it's more readable with all fields specified it's ok to leave them.